compiler/src/iree/compiler/Codegen/SPIRV/SPIRVTile.cpp

```diff
@@ -143,37 +146,34 @@ static LogicalResult tileAndDistributeToThreads(linalg::LinalgOp consumerOp,
 }
 
 /// Tiles reduction dimensions.
-static LogicalResult tileReduction(func::FuncOp funcOp,
-                                   ArrayRef<int64_t> tileSizes) {
+static LogicalResult
+tileReduction(func::FuncOp funcOp,
+              const scf::SCFTileSizeComputationFunction &computeFn) {
   MLIRContext *context = funcOp.getContext();
-
-  SmallVector<Operation *> candidates;
+  // Set markers to drive tiling reduction dimensions.
+  OpBuilder builder(context);
+  auto marker = builder.getStringAttr(getTileReductionMarker());
   funcOp.walk([&](linalg::LinalgOp op) {
-    if (isa<linalg::ContractionOpInterface, linalg::ConvolutionOpInterface,
-            linalg::GenericOp>(op.getOperation())) {
-      candidates.push_back(op);
+    if (isa<linalg::ContractionOpInterface>(*op) ||
+        isa<linalg::ConvolutionOpInterface>(*op) ||
+        isa<linalg::GenericOp>(*op)) {
+      op->setAttr(IREE::LinalgExt::LinalgTransforms::kLinalgTransformMarker,
+                  marker);
     }
   });
 
-  IRRewriter rewriter(context);
-  for (auto op : candidates) {
-    auto target = cast<TilingInterface>(op);
-    scf::SCFTilingOptions options;
-    setSCFTileSizes(options, target, tileSizes, /*tileScalableFlags=*/{});
-    FailureOr<scf::SCFTilingResult> tiledResults =
-        scf::tileUsingSCFForOp(rewriter, target, options);
-    if (failed(tiledResults)) {
-      return failure();
-    }
-    rewriter.replaceOp(op, tiledResults->replacements);
-  };
+  auto filter =
+      IREE::LinalgExt::LinalgTransformationFilter({marker}, std::nullopt);
```

Comment:
This part is as same as https://github.com/openxla/iree/commit/9b74e956dafd5b11c5581943b4117bbc38eda115#diff-b8b284fc638e6a2c004e1c0102c7a41699d1a2b6bbc9d31b46ea79fdabd64b8aL183-L193